[JustinSoh] iP

JustinSoh opened the pull request.

JustinSoh committed:
Initial structure of mimichat

- Initial structure implemented

JustinSoh committed:
Add Echo

- Added ability for bot to echo
- bot able to say bye

JustinSoh committed:
Add List

- Change to switch statements
- Able to add to taskList
- Able to print out taskList

JustinSoh committed:
Add Mark and Unmark

- Added Task Class
- Mark and Unmark feature done

JustinSoh committed:
Clean up Coding Standard

JustinSoh committed:
add test case

- added test case for level 3

JustinSoh committed:
Add ToDos, Events, Deadlines

- Added Subclasses for Task

JustinSoh committed:
Updated testcase

LuoYu-uwu commented on src/main/java/Task.java:
```diff
@@ -0,0 +1,43 @@
+        this.name = name;
+    }
+
+    public void setDone(boolean done) {
```
boolean done should be named isDone to sound like booleans

JustinSoh replied:
ok will make the neccessary changes. ty

LuoYu-uwu commented on src/main/java/Duke.java:
```diff
@@ -1,10 +1,163 @@
+        while (isRunning) {
+            String input = scanner.nextLine();
+            String[] inputs = input.split(" ", 2);
+            switch (inputs[0]) {
```
avoid magic numbers

JustinSoh replied:
noted

LuoYu-uwu commented on src/main/java/Duke.java:
```diff
@@ -1,10 +1,163 @@
+                shutdownSequence();
+                break;
+            case "list":
+                listTasks(taskList, numberOfTask);
```
good use of plural naming

annnniexu commented on src/main/java/Duke.java:
```diff
@@ -1,10 +1,163 @@
+            String input = scanner.nextLine();
+            String[] inputs = input.split(" ", 2);
+            switch (inputs[0]) {
+            case "bye":
```
switch statement makes it easy to read

LuoYu-uwu commented on src/main/java/Duke.java:
```diff
@@ -1,10 +1,163 @@
+import java.util.Scanner;
+
 public class Duke {
+    final static String APP_NAME = "mimichat";
```
good use of final modifier and using capital letters for caps

annnniexu commented on src/main/java/Duke.java:
```diff
@@ -1,10 +1,163 @@
 public class Duke {
+    final static String APP_NAME = "mimichat";
+    static Task[] taskList = new Task[100];
+    static int numberOfTask = 0;
```
could use "taskCount or numberOfTasks"

annnniexu commented on src/main/java/Duke.java:
```diff
@@ -1,10 +1,163 @@
+        String endDate = duration[1].strip();
+
+        // Add into task list and print success message
+        Event event = new Event(taskName, startDate, endDate);
```
addDeadline and addEvent have similar ending code and thus could be consolidated

annnniexu commented on text-ui-test/EXPECTED.TXT:
```diff
@@ -1,7 +1,50 @@
-| |_| | |_| |   <  __/
-|____/ \__,_|_|\_\___|
-
+-------------------------------------------
```
great code and great tests! overall very easy to read

JustinSoh committed:
Add InvalidInput Format Exception

- to throw an exception when the input length is too short

JustinSoh committed:
Update InvalidInput Exception

updated certain modules

JustinSoh committed:
Auto stash before rebase of "refs/heads/branch-Level-5"

JustinSoh committed:
Auto stash before rebase of "refs/heads/branch-Level-5"

JustinSoh committed:
fixed

JustinSoh committed:
add invalid instruction exception

JustinSoh committed:
add exception handling for events

- included test for it as well

JustinSoh committed:
add exception handling for mark/unmark

- include code tidy up

JustinSoh committed:
Merge branch 'branch-Level-5'

JustinSoh committed:
add packages

JustinSoh committed:
updated

JustinSoh committed:
Merge branch 'branch-A-Packages'

json-bot-boop commented on src/main/java/Duke.java:
```diff
@@ -1,10 +1,289 @@
+    final static String APP_NAME = "mimichat";
+    final static int INPUT_LIMIT = 2;
+
+    static Task[] taskList = new Task[100];
```
Try to avoid magic number

json-bot-boop commented on src/main/java/Duke.java:
```diff
@@ -1,10 +1,289 @@
+     * Main method to run the program
+     *
+     * @param args
+     */
```
Try not to write comments information that is obvious in the code. This code quality will be taught in future lectures.

json-bot-boop commented on src/main/java/Duke.java:
```diff
@@ -1,10 +1,289 @@
+            if (inputs.length < 2) {
+                throw new InsufficientParameters(MimiException.INSUFFICIENT_DEADLINE_PARAMETERS);
+            }
+            String[] splitInputs = inputs[1].split("/by", 2);
```
Try to avoid magic string

json-bot-boop commented on src/main/java/Duke.java:
```diff
@@ -1,10 +1,289 @@
+        }
+    }
+
+    private static int checkValidityOfMarks(String[] inputs) throws TaskNotFound,
```
Try to define your method before using them

json-bot-boop commented on src/main/java/Duke.java:
```diff
@@ -1,10 +1,289 @@
+        } catch (InsufficientParameters | IncorrectFormat e) {
+            System.out.println(e.getMessage());
+        }
+    }
```
Can consider the case where the argument value is empty

json-bot-boop reviewed (commented):
Overall, it is good except some code quality issue. Keep it up

JustinSoh committed:
cleanup

JustinSoh committed:
swap to arraylist

- tested with testcase and all works

JustinSoh committed:
add delete

- included delete function
- included test cases

JustinSoh committed:
add saveFile()

able to save file into data/mimi.chat

JustinSoh committed:
modified test case to work with files

JustinSoh committed:
add exception for corrupted files

JustinSoh committed:
Merge branch 'branch-Level-6'

JustinSoh committed:
Merge branch 'branch-Level-7'

JustinSoh committed:
perform merge cleanup

JustinSoh committed:
generated jar file

JustinSoh committed:
Changed delimiter to '|'

- found a bug with the use of comma
- added test case with the bugs
- removed magic strings

Commit:
added ui

created ui package for dealing with user interface methods

Commit:
Added storage and tasklist

Commit:
added Parser

JustinSoh committed:
add find functionality

- able to find
- test cases updated too

JustinSoh committed:
Merge branch 'branch-Level-9'

JustinSoh committed:
Merge pull request #1 from JustinSoh/branch-Level-9

add find functionality

JustinSoh committed:
Merge branch 'master' of https://github.com/JustinSoh/ip

JustinSoh committed:
Add Java Docs

JustinSoh committed:
Merge pull request #2 from JustinSoh/A-JavaDoc

Add Java Docs

JustinSoh committed:
added user guide

JustinSoh committed:
Merge pull request #3 from JustinSoh/PublishedUG

added user guide

JustinSoh committed:
tweaks to user guide

JustinSoh committed:
add find exception

Commit:
found bugs and updated accordingly

Commit:
add directory creation bug

Commit:
Merge branch 'master' into PublishedUG

Commit:
Merge branch 'PublishedUG'
